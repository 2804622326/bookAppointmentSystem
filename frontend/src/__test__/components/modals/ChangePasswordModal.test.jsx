import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import ChangePasswordModal from '../../../components/modals/ChangePasswordModal.jsx';

jest.mock('../../../components/user/UserService', () => ({
  changeUserPassword: jest.fn().mockResolvedValue({ message: 'ok' }),
}));

test('submits password change', async () => {
  const { changeUserPassword } = require('../../../components/user/UserService');
  const { asFragment } = render(
    <ChangePasswordModal userId={1} show handleClose={() => {}} />
  );
  fireEvent.change(screen.getByLabelText(/current password/i), { target: { value: 'a' } });
  fireEvent.change(screen.getByLabelText(/^new password/i), { target: { value: 'b' } });
  fireEvent.change(screen.getByLabelText(/confirm new password/i), { target: { value: 'b' } });
  fireEvent.click(screen.getByRole('button', { name: /change password/i }));
  await waitFor(() => expect(changeUserPassword).toHaveBeenCalled());
  expect(asFragment()).toMatchSnapshot();
});
