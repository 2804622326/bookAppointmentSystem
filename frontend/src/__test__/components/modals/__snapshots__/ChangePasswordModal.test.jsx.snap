// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`submits password change 1`] = `<DocumentFragment />`;
